NYULibraries/ariadne
# Add support for institutions: additions

monday.com: [Add support for institutions to Ariadne frontend](https://nyu-lib.monday.com/boards/765008773/pulses/3778586350?userId=27226106)

* _e2e/.env.test_
	* Remove vestigial `REACT_APP_PUBLIC_URL`
* _frontent/src/aux/institutionConstants.js_ renamed to _institutionInfo.js_
  	* Rename `institutions` to `bannerInstitutionInfo` and `deepFreeze` it
* _frontent/src/aux/helpers.js_
	* Use `function` to declare functions instead of `const` with arrow function expressions
	* Add MDN's `deepFreeze` function
	* Alpha-sort functions
	* Hardcode "institution" in `getInstitutionQueryParameter` instead of passing it in as an argument
	* Make `getParameterFromQueryString` 100% case-insensitive (param name can be any case) and inline `parameter`
* _frontent/src/aux/helpers.test.js_
	* Parameterize tests in `describe.each` loop
	* Add tests for all case combinations of query param name and value
* _frontend/components/Banner.js_
	* Inline `bannerInstitutionInfo`
* _frontend/components/Banner.test.js_
	* Parameterize tests in `describe.each` loop
	* Add test for `institution=nyu` `<img>` `class` attribute
	* Update snapshots
	* Change `window.history.pushState` to `window.location` replacement method used in `List.test.js`.  See [commit message](https://github.com/NYULibraries/ariadne/pull/78/commits/2e2c980ae6d6d3fe4d853089ab86a5d20fb8dfea) for reasons.
